common/src/main/java/com/microsoft/identity/common/internal/result/ResultFuture.java

```diff
@@ -28,21 +59,55 @@ public boolean isDone() {
     @Override
     public T get() throws InterruptedException {
         mCountDownLatch.await();
+
+        if (null != mException) {
+            throw new RuntimeException(mException);
+        }
+
         return mResult;
     }
 
     @Override
-    public T get(long l, TimeUnit timeUnit) throws InterruptedException, TimeoutException {
+    public T get(final long l, @NonNull final TimeUnit timeUnit) throws InterruptedException, TimeoutException {
```

Comment:
I was looking at this, I thought Future threw ExecutionExcepttion as well.  If it does, that's a better throw than runtime below.

Reply:
fae08476ddf46ae797c5adfd62f1632a4190014a

Adding this checked Exception will require another PR to add `try`/`catch` blocks to MSAL so keep an eye out for that once this merges